### ISSUE TITLE
[Backport v2.7.6] [BUG] RKE2 provisioning with custom NodeDriver fails when rancher has more than one replica

### ISSUE THREAD

**Issue by rancherbot** [rancher/rancher#42128]
This is a backport issue for https://github.com/rancher/rancher/issues/38481, automatically created via rancherbot by @Sahota1225

Original issue description:

**Rancher Server Setup**
- Rancher version: 2.6.6
- Installation option (Docker install/Helm Chart): Helm Chart

**Information about the Cluster**
- Kubernetes version: RKE2  v1.23.7
- Cluster Type (Local/Downstream): Local

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**

When trying to provision rke2 node with custom docker machine driver (eg. [docker-machine-driver-proxmox-ve](https://github.com/lnxbil/docker-machine-driver-proxmox-ve)), provisioning job fails as it can not find the driver in /assets.
I found out that the driver appears in the /assets directory but only in one rancher instance instead of all in deployment.

**To Reproduce**
Provision rke2 cluster with custom node driver on rancher with more than 1 instance running. 
Putting driver manualy on another instances to /usr/share/rancher/ui/assets/ fixes the issue.

**Result**
```Status:
  Cloud Credential Secret Name:  cattle-global-data:cc-fm9vm
  Conditions:
    Message:  failed creating server [fleet-default/am4-rke2-test-pool1-36e4ded7-r72zm] of kind (ProxmoxveMachine) for machine am4-rke2-test-pool1-77bd49cbf5-sg922 in infrastructure provider: CreateError: Downloading driver from https://rancher.xxx/assets/docker-machine-driver-proxmoxve
Doing /etc/rancher/ssl
downloaded file docker-machine-driver-proxmoxve failed sha256 checksum
download of driver from https://rancher.xxx/assets/docker-machine-driver-proxmoxve failed
    Reason:         CreateError
    Status:         False
    Type:           CreateJob
    Message:        deleting server [fleet-default/am4-rke2-test-pool1-36e4ded7-r72zm] of kind (ProxmoxveMachine) for machine am4-rke2-test-pool1-77bd49cbf5-sg922 in infrastructure provider
    Reason:         
    Status:         False
    Type:           Ready
  Driver Hash:      dd28695b5c454703e9cdb51b0e39719001d8927815f91a03bd1326843fffbd18
  Driver URL:       https://rancher.xxx/assets/docker-machine-driver-proxmoxve
  Failure Message:  Downloading driver from https://rancher.xxx/assets/docker-machine-driver-proxmoxve
Doing /etc/rancher/ssl
downloaded file docker-machine-driver-proxmoxve failed sha256 checksum
download of driver from https://rancher.xxx/assets/docker-machine-driver-proxmoxve failed
```

**Expected Result**
Node should start provisioning and the drivers should appear in /ui/assets in all rancher instances. 

**Screenshots**

**Additional context**
I found out that some of drivers are loaded in dockerfile, probably as the workaround for this issue (https://github.com/rancher/rancher/blob/release/v2.6/package/Dockerfile#L72-L83)

SURE-5893

**Comment by markusewalker** [comment]
**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|Setup Rancher with 1 replica using Proxmox VE, provision a downstream RKE2 node driver cluster as a standard user, validate that the docker machine driver appears in the /assets directory|
|2|Setup Rancher with 3 replicas using Proxmox VE, provision a downstream RKE2 node driver cluster as a standard user, validate that the docker machine driver appears in the /assets directory|

It's important to note issues https://github.com/rancher/rancher/issues/37074 and https://github.com/rancher/dashboard/issues/8563 are preventing QA from adding Proxmox VE as a custom node driver directly into Rancher. Therefore, the above test plan takes into account these issues.

**Comment by markusewalker** [comment]
Reproduced this on `v2.7.5`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Helm
- Rancher version: `v2.7.5`
- Browser: Chrome
- Proxmox VE

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|Setup Rancher with 3 replicas, add custom driver Proxmox VE, provision a downstream RKE2 custom node driver cluster as a standard user, validate that the docker machine driver appears in the /assets directory|:x:|

**REPRO STEPS**

**Scenario 1**
1. Setup a separate Proxmox server and a separate Rancher server with 3 replicas.
2. As an admin user, added custom node [driver](https://github.com/lnxbil/docker-machine-driver-proxmox-ve) for Proxmox VE.
     - Due to this [issue](https://github.com/rancher/rancher/issues/37074), I had to run the following command in my client machine: `kubectl get nodedriver <listed Proxmox name> -o yaml > driver.yaml`
     - Excluded everything underneath `metadata` except name and inputted `proxmoxve`. Additionally, removed the `status` block.
     - Ran command `kubectl apply -f driver.yaml`.
3. Noted that the correct Proxmox VE shows in the node driver list in the Rancher UI.
4. Added the following annotations to the custom node driver:
```
privateCredentialFields: proxmoxUserPassword
publicCredentialFields: proxmoxUserName
```
5. As a standard user, attempted to provision a downstream RKE2 `v1.26.6+rke2r1` custom node driver cluster and was unable to.
6. Noted as well that  the driver DID NOT appear in the `/assets` directory as originally reported.

Now that this is reported, QA will attempt to test on the latest `v2.7-head` and see if this issue has been addressed. A Rancher RC will need to be cut, however, for testing to commence.

**Comment by markusewalker** [comment]
For transparency for this issue, QA will be moving away from Proxmox VE and over to Nutanix for testing. As such, the test plan has been updated to accomodate this change:

**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|Setup Rancher with 3 replicas, add custom driver Nutanix, provision a downstream RKE2 node driver cluster (1 etcd, 1 cp, 1 worker) as a standard user|
|2|Setup Rancher with 3 replicas, add custom driver Nutanix, provision a downstream RKE2 node driver cluster (1 etcd + 1 cp, 1 worker) as a standard user, upgrade the K8s version|
|3|Setup Rancher v2.7.5 with 3 replicas, add custom driver Nutanix, provision a downstream K3S node driver cluster (1 etcd, 1 cp, 1 worker) as a standard user, upgrade to Rancher v2.7-head, provision a new downstream K3S node driver cluster as a standard user|

**Comment by markusewalker** [comment]
Reproduced this on `v2.7.5`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Helm
- Rancher version: `v2.7.5`
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|Setup Rancher with 3 replicas, add custom driver Nutanix, provision a downstream RKE2 custom node driver cluster as a standard user|:x:

**VALIDATION STEPS**

**Scenario 1**
1. Setup Rancher via Helm and ensured 3 replicas are available.
2. As a standard user, attempted to provision a downstream RKE2 Nutanix node driver `v1.26.7+rke2r1`  cluster (1 etcd, 1 cp, 1 worker).
3. Received the following error message:
![image](https://github.com/rancher/rancher/assets/52591819/06346d91-b7d7-4843-8ee3-f36dfe4cc782)

Now that this is reproduced, I will be testing on the latest `v2.7.6-head` to validate that this is addressed.

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.7.6` https://github.com/rancher/rancher/commit/68bd3be1b794cc9ec5473e4187b31dd92eeebb6a. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Helm
- Rancher version: `v2.7.6-head` https://github.com/rancher/rancher/commit/68bd3be1b794cc9ec5473e4187b31dd92eeebb6a
- Browser: Chrome

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|Setup Rancher with 3 replicas, add custom driver Nutanix, provision a downstream RKE2 custom node driver cluster as a standard user|:white_check_mark:|
|2|Setup Rancher with 3 replicas, add custom driver Nutanix, provision a downstream RKE2 custom node driver cluster as a standard user, upgrade the K8s version|:white_check_mark:|
|3|Setup Rancher v2.7.5 with 3 replicas, add custom driver Nutanix, provision a downstream RKE2 custom node driver cluster as a standard user, upgrade to Rancher v2.7.6-head, provision a new downstream RKE2 custom node driver cluster as a standard user|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. Setup Rancher via Helm and ensured 3 replicas are available.
2. As a standard user, provisioned a downstream RKE2 Nutanix node driver `v1.26.7+rke2r1`  cluster (1 etcd, 1 cp, 1 worker).
3. Noted that the cluster is stuck at `Waiting for cluster agent to connect`, but the reported issue has been addressed.
     - Root causing the issue has shown this to be a DNS issue with Nutanix as opposed to a Rancher issue.
     - As such, marking this test case as passing as this unrelated DNS issue is causing the cluster to not be `Active`.
4. Repeated this scenario using a Docker install instead of Helm; the cluster and nodes DID come up as `Active`.
     - The only difference was the Rancher server URL used the corresponding IP address and not it's specified DNS. This confirms that the DNS is the issue.
________

**Scenario 2**
1. Repeated scenario 1, but provisioned a downstream RKE2 Nutanix node driver `v1.25.17+rke2r1` cluster (1 etcd + cp, worker).
2. Received the same results as scenario 1.
3. Upgraded the downstream cluster to `v1.26.7+rke2r1`.
4. Ran into the above-mentioned DNS issue, so the cluster remained in `Waiting for cluster agent to connect` status.
5. Repeated this scenario using a Docker install instead of Helm; the cluster and nodes DID come up as `Active`.
     - The only difference was the Rancher server URL used the corresponding IP address and not it's specified DNS. This confirms that the DNS is the issue.

__________

**Scenario 3**
1. Setup Rancher `v2.7.5` via Helm and ensured 3 replicas are available.
2. As a standard user, provisioned a downstream RKE2 Nutanix node driver `v1.26.7+rke2r1`  cluster (1 etcd, 1 cp, 1 worker).
3. Noted that the cluster is stuck at `Waiting for cluster agent to connect`, but the reported issue has been addressed.
4. Upgraded Rancher to `v2.7.6-head`; validated that the cluster did not auto-upgrade the K8s version.
5. Provisioned a new downstream RKE2 Nutanix node driver `v1.26.7+rke2r1`  cluster (1 etcd, 1 cp, 1 worker).
     - Noted the same DNS issue as seen, leaving the cluster in state `Waiting for cluster agent to connect`.
